include/oneapi/dpl/pstl/hetero/numeric_impl_hetero.h

```diff
@@ -91,23 +102,249 @@ __pattern_transform_reduce(_ExecutionPolicy&& __exec, _ForwardIterator __first,
+                        }
+                    }
+                    else
+                    {
+                        _ONEDPL_PRAGMA_UNROLL
+                        for (::std::uint16_t __idx = __item_id; __idx < __n; __idx += _WGSize)
+                        {
+                            __out[__idx] = __lacc[__idx];
+                        }
+
+                        const ::std::uint16_t __residual = __n % _WGSize;
+                        const ::std::uint16_t __residual_start = __n - __residual;
+                        if (__item_id < __residual)
+                        {
+                            auto __idx = __residual_start + __item_id;
+                            __out[__idx] = __lacc[__idx];
+                        }
+                    }
+                });
+        });
+        __event.wait();
+    }
+
+    template <typename _ValueType, typename _Group, typename _Begin, typename _End, typename _BinaryOperation>
+    static void
+    __group_scan(const _Group& __group, _Begin __begin, _End __end, _BinaryOperation __bin_op,
+                 unseq_backend::__no_init_value<_ValueType>)
+    {
+        if constexpr (_Inclusive)
+            __dpl_sycl::__joint_inclusive_scan(__group, __begin, __end, __begin, __bin_op);
+        else
+            __dpl_sycl::__joint_exclusive_scan(__group, __begin, __end, __begin, __bin_op);
+    }
+
+    template <typename _ValueType, typename _Group, typename _Begin, typename _End, typename _BinaryOperation>
+    static void
+    __group_scan(const _Group& __group, _Begin __begin, _End __end, _BinaryOperation __bin_op,
+                 unseq_backend::__init_value<_ValueType> __init)
+    {
+        if constexpr (_Inclusive)
+            __dpl_sycl::__joint_inclusive_scan(__group, __begin, __end, __begin, __bin_op, __init.__value);
+        else
+            __dpl_sycl::__joint_exclusive_scan(__group, __begin, __end, __begin, __init.__value, __bin_op);
+    }
+};
+
 //------------------------------------------------------------------------
 // transform_scan
 //------------------------------------------------------------------------
-
 template <typename T>
 struct ExecutionPolicyWrapper;
 
-template <typename _ExecutionPolicy, typename _Iterator1, typename _Iterator2, typename _UnaryOperation,
-          typename _InitType, typename _BinaryOperation, typename _Inclusive>
+template <typename _ExecutionPolicy, typename _InRng, typename _OutRng, typename _UnaryOperation, typename _InitType,
+          typename _BinaryOperation, typename _Inclusive>
 void
-__pattern_transform_scan_base_impl(_ExecutionPolicy&& __exec, _Iterator1 __first, _Iterator1 __last,
-                                   _Iterator2 __result, _UnaryOperation __unary_op, _InitType __init,
-                                   _BinaryOperation __binary_op, _Inclusive)
+__pattern_transform_scan_single_group(_ExecutionPolicy&& __exec, _InRng&& __in_rng, _OutRng __out_rng,
```

Comment:
That pattern should be moved to following file:
https://github.com/oneapi-src/oneDPL/blob/6cd24948245a409eb6c2f61c2d07af43471f0a43/include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

Reply:
Fixed. It's now in parallel_backend_sycl.h.